src/OrchardCore.Modules/OrchardCore.PublishLater/Services/ScheduledPublishingBackgroundTask.cs

```diff
@@ -10,37 +10,41 @@
 using OrchardCore.PublishLater.Indexes;
 using YesSql;
 
-namespace OrchardCore.PublishLater.Services
+namespace OrchardCore.PublishLater.Services;
+
+[BackgroundTask(Schedule = "* * * * *", Description = "Publishes content items when their scheduled publish date time arrives.")]
+public class ScheduledPublishingBackgroundTask : IBackgroundTask
 {
-    [BackgroundTask(Schedule = "* * * * *", Description = "Publishes content items when their scheduled publish date time arrives.")]
-    public class ScheduledPublishingBackgroundTask : IBackgroundTask
+    private readonly ILogger<ScheduledPublishingBackgroundTask> _logger;
+    private readonly IClock _clock;
+
+    public ScheduledPublishingBackgroundTask(ILogger<ScheduledPublishingBackgroundTask> logger, IClock clock)
+    {
+        _logger = logger;
+        _clock = clock;
+    }
+
+    public async Task DoWorkAsync(IServiceProvider serviceProvider, CancellationToken cancellationToken)
     {
-        private readonly ILogger<ScheduledPublishingBackgroundTask> _logger;
-        private readonly IClock _clock;
+        var itemsToPublish = await serviceProvider
+            .GetRequiredService<ISession>()
+            .QueryIndex<PublishLaterPartIndex>(index => index.Latest && !index.Published && index.ScheduledPublishDateTimeUtc < _clock.UtcNow)
+            .ListAsync();
 
-        public ScheduledPublishingBackgroundTask(ILogger<ScheduledPublishingBackgroundTask> logger, IClock clock)
+        if (!itemsToPublish.Any())
         {
-            _logger = logger;
-            _clock = clock;
+            return;
         }
 
-        public async Task DoWorkAsync(IServiceProvider serviceProvider, CancellationToken cancellationToken)
+        var contentManager = serviceProvider.GetRequiredService<IContentManager>();
+
+        foreach (var item in itemsToPublish)
         {
-            var itemsToPublish = await serviceProvider
-                .GetRequiredService<ISession>()
-                .Query<ContentItem, PublishLaterPartIndex>(index => index.ScheduledPublishDateTimeUtc < _clock.UtcNow)
-                .ListAsync();
-
-            if (!itemsToPublish.Any())
-            {
-                return;
-            }
-
-            foreach (var item in itemsToPublish)
-            {
-                _logger.LogDebug("Publishing scheduled content item {ContentItemId}.", item.ContentItemId);
-                await serviceProvider.GetRequiredService<IContentManager>().PublishAsync(item);
-            }
+            var contentItem = await contentManager.GetAsync(item.ContentItemId);
```

Comment:
So here we may want to set the schedule date to null, and then remove the related handler, this if we use the same logic as in ArchiveLater.